src/Processors/Formats/IRowInputFormat.cpp

```diff
@@ -104,6 +106,30 @@ Chunk IRowInputFormat::generate()
             }
             catch (Exception & e)
             {
+                /// Record error info for this row
+                String diagnostic;
+                String raw_data;
+                try
+                {
+                    std::tie(diagnostic, raw_data) = getDiagnosticAndRawData();
+                }
+                catch (const Exception & exception)
+                {
+                    diagnostic = "Cannot get diagnostic: " + exception.message();
+                    raw_data = "Cannot get raw data: " + exception.message();
+                }
+                catch (...)
+                {
+                    /// Error while trying to obtain verbose diagnostic. Ok to ignore.
+                }
+                trimLeft(diagnostic, '\n');
+                trimRight(diagnostic, '\n');
+
+                auto now_time = time(nullptr);
+
+                if (errors_logger)
+                    errors_logger->logError(InputFormatErrorsLogger::ErrorEntry{to_string(now_time), total_rows, diagnostic, raw_data});
```

Comment:
Let's move it after check `if (params.allow_errors_num == 0 && params.allow_errors_ratio == 0)`, because it make sense to record errors only when this check is false. Also I would make a new method `logError` in `IRowInputFormat` with this logic and call it only when `errors_logger` is set.